models/evc.py

```diff
@@ -1410,6 +1410,63 @@ def handle_link_down(self):
 
         return success
 
+    @staticmethod
+    def get_interface_from_switch(uni: UNI, switches: dict) -> Interface:
+        """Get interface from switch by uni"""
+        switch = switches[uni.interface.switch.dpid]
+        interface = switch.interfaces[uni.interface.port_number]
+        return interface
+
+    def handle_topology_update(self, switches: dict):
+        """Handle changes in the topology"""
+        try:
+            interface_a = self.get_interface_from_switch(self.uni_a, switches)
+        except KeyError:
+            id_ = self.uni_a.interface.id
+            log.warning(f"The interface {id_} was not found")
+            return
+
+        try:
+            interface_z = self.get_interface_from_switch(self.uni_z, switches)
+        except KeyError:
+            id_ = self.uni_z.interface.id
+            log.warning(f"The interface {id_} was not found")
+            return
+
+        active, interface = self.is_uni_interface_active(
+            interface_a, interface_z
+        )
+        if self.is_active() != active:
```

Comment:
There's a bug in the activation logic, if you were to disable all uplinks, it ended up still activating an EVC that only had its UNI UP (notice below the EVC not being deployed, but was still activated), for instance with the ring topology if you were to shutdown `s1-eth4` and `s1-eth3`, while having an EVC between eth1 S1 - S3 eth3, could you double check and fix this?

```
kytos $> 2023-07-12 17:04:48,873 - INFO [kytos.napps.kytos/of_core] [main.py:708:update_port_status] (MainThread) PortStatus modified interface 00:00:00:00:00:00:00:01:3 state 0        
2023-07-12 17:04:48,874 - INFO [kytos.napps.kytos/of_core] [main.py:708:update_port_status] (MainThread) PortStatus modified interface 00:00:00:00:00:00:00:01:3 state OFPPS_LINK_DOWN
2023-07-12 17:04:48,875 - INFO [kytos.napps.kytos/of_core] [main.py:708:update_port_status] (MainThread) PortStatus modified interface 00:00:00:00:00:00:00:02:2 state 0
2023-07-12 17:04:48,875 - INFO [kytos.napps.kytos/of_core] [main.py:708:update_port_status] (MainThread) PortStatus modified interface 00:00:00:00:00:00:00:02:2 state OFPPS_LINK_DOWN
2023-07-12 17:04:48,878 - INFO [kytos.napps.kytos/mef_eline] [main.py:737:handle_link_down] (thread_pool_app_10) Event handle_link_down Link(Interface('s1-eth3', 3, Switch('00:00:00:00:
00:00:00:01')), Interface('s2-eth2', 2, Switch('00:00:00:00:00:00:00:02')), 78282c4d5b579265f04ebadc4405ca1b49628eb1d684bb45e5d0607fa8b713d0)
2023-07-12 17:04:48,885 - INFO [kytos.napps.kytos/flow_manager] [main.py:624:_send_flow_mods_from_request] (AnyIO worker thread) Send FlowMod from request dpid: 00:00:00:00:00:00:00:01,
 command: delete, force: True, flows_dict: {'flows': [{'cookie': 12313818680397910347, 'cookie_mask': 18446744073709551615}], 'force': True}
2023-07-12 17:04:48,893 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:36502 - "POST /api/kytos/flow_manager/v2/delete/00%3A00%3A00%3A00%3A00%3A00%3A00%3A01
 HTTP/1.1" 202
2023-07-12 17:04:48,902 - INFO [kytos.napps.kytos/flow_manager] [main.py:624:_send_flow_mods_from_request] (AnyIO worker thread) Send FlowMod from request dpid: 00:00:00:00:00:00:00:02,
 command: delete, force: True, flows_dict: {'flows': [{'cookie': 12313818680397910347, 'cookie_mask': 18446744073709551615}], 'force': True}
2023-07-12 17:04:48,909 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:36512 - "POST /api/kytos/flow_manager/v2/delete/00%3A00%3A00%3A00%3A00%3A00%3A00%3A02
 HTTP/1.1" 202
2023-07-12 17:04:48,916 - INFO [kytos.napps.kytos/flow_manager] [main.py:624:_send_flow_mods_from_request] (AnyIO worker thread) Send FlowMod from request dpid: 00:00:00:00:00:00:00:03,
 command: delete, force: True, flows_dict: {'flows': [{'cookie': 12313818680397910347, 'cookie_mask': 18446744073709551615}], 'force': True}
2023-07-12 17:04:48,925 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:36526 - "POST /api/kytos/flow_manager/v2/delete/00%3A00%3A00%3A00%3A00%3A00%3A00%3A03
 HTTP/1.1" 202
2023-07-12 17:04:48,959 - INFO [kytos.napps.kytos/pathfinder] [main.py:101:shortest_path] (AnyIO worker thread) POST v2/ payload data: {'source': '00:00:00:00:00:00:00:01:1', 'destinati
on': '00:00:00:00:00:00:00:03:1', 'spf_max_paths': 2}
2023-07-12 17:04:48,960 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:36542 - "POST /api/kytos/pathfinder/v3/ HTTP/1.1" 200
2023-07-12 17:04:48,961 - WARNING [kytos.napps.kytos/mef_eline] [evc.py:738:deploy_to_path] (thread_pool_app_1) EVC(e378d931afd14b, epl) was not deployed. No available path was found.
2023-07-12 17:04:48,971 - INFO [kytos.napps.kytos/mef_eline] [evc.py:1442:handle_topology_update] (thread_pool_app_7) Activating EVC e378d931afd14b
```

Reply:
Thanks. Turned out that in the cases where the handling of `link_down` finished, the event `topology_update` was called. Sine `link_down` also deletes current path, added a condition to check it when handling `topology_update`.

Reply:
Right. Looks good. Let's close this thread.